### DISCUSSION TITLE
Device configuration backup

### DISCUSSION THREAD

**Discussion by mitch2k** [netbox-community/netbox#11274] [General]
Hi,
I'm currently using Netbox as DCIM and rConfig v3 as config backup tool. With rConfig we get a config backup twice a day, and we get an e-mail if the config has been changed.
I'm wondering if there is a (solid) way to implement the same functionality in Netbox so we can phase-out rConfig. Googling around I did find some plugins, but looks like the where not capable of alerting in case of a config change.
Any ideas?
Thanks!

**Comment by kagarcia1618** [comment]
My recommendation is not using a Netbox plugin but instead another 3rd party solution using gitlab. You may want to explore integrating your Netbox with a private gitlab server using a python script. You can refer to my article here: `https://github.com/kagarcia1618/Network-Config-Tracker`

**Comment by xkilian** [comment]
I would suggest using an ansible playbook with an inventory produced by netbox.
The cisco module for connecting via ssh is an excellent generic method for getting the config and show commands and commit them to a git. Your playbook should also send backup state back to your monitoring system of choice(ex prometheus pushgateway).
As stated, this functionality should be external to netbox. It does not need to interact with netbox in any meaningful way other than getting an inventory of devices, they model, os and if the device is considered active (in the field).
That is what we use. A dignified successor to old rancid. Including diffs and emails.
You can then do post processing to extract relevant reports like mac/ip port mapping, serial number inventories for support contracts and other types of reports.
That is why you should collect config backups AND show commands to git.
A little internet sleuthing should provide a basic ansible playbook with the cisco module to get you started.
Cheers.
